### DISCUSSION TITLE
What is the recommended recipe for running VoucherVision entirely locally for free?

### DISCUSSION THREAD

**Discussion by Gene-Weaver** [Gene-Weaver/VoucherVision#10] [Q&A / FAQs]
VoucherVision supports several local implementations and workflows (with more on the way).

### **Option A: Run OCR and LLM locally**
_Hardware requirements:_
- At least 100 GB of free SSD storage space to hold the LLMs.

- You will need an Nvidia GPU with at least 24 GB of VRAM such as:
  - RTX Titan
  - RTX 4090
  - RTX 4500 Ada 
  - RTX A6000 
  - RTX 6000 Ada
  - H100
  - A100

_Local OCR_ 

- We recommend using `Florence-2` and `Google Vision Handwritten` simultaneously, but Florence-2 works well on its own for most labels. 

_Local LLM_ 
- Select one of the models from the dropdown menu that has the `LOCAL` prefix. Right now we recommend `LOCAL Mistral 7B Instruct v0.2` because it is small and capable. If your GPU has 40GB+ of VRAM then you can use `LOCAL Mi**x**tral ...` versions. 

### **Option B: Use Google Vision API for OCR and and run LLM locally**
_Hardware requirements:_
- At least 100 GB of free SSD storage space to hold the LLMs.

- You will need an Nvidia GPU with at least 16 GB of VRAM, but ideally 24GB of VRAM such as:
  - RTX 4080
  - RTX 4090
  - RTX 4000
  - RTX A4500
  - RTX 4500 Ada 
  - RTX A6000 
  - RTX 6000 Ada
  - H100
  - A100

_OCR_ 
- Use `Google Vision Handwritten`

_Local LLM_ 
- Select one of the models from the dropdown menu that has the `LOCAL` prefix. Right now we recommend `LOCAL Mistral 7B Instruct v0.2` because it is small and capable. If your GPU has 40GB+ of VRAM then you can use `LOCAL Mi**x**tral ...` versions. 

### Notes:

- As of July 17, 2024 Florence-2 returns OCR text that often lacks spaces, i.e. has words touchinglikethis. So after Florence-2 we send the text to a 4-bit version of mistral-7b to correct minor punctuation and formatting errors, so this takes up about 4GB of VRAM extra.
- As of July 17, 2024 Option A has only been validated on systems that have at least 24GB of VRAM. Please let us know what you have found to work or not work.

**Comment by nickynicolson** [comment]
At Kew, we developed a (skeletal) set of notes about installing Voucher Vision on a shared high performance cluster (one available to Kew, Natural History Museum, London and Royal Botanic Gardens Edinburgh researchers). Posted as these might be useful for anybody else who wants to run the web interface to voucher vision in an HPC environment (requires port forwarding to access the interface): https://github.com/vvatk/.github/blob/main/docs/VV-install-on-HPC.md
